src/io/iohk/scalanet/test/peergroup/DTLSPeerGroupSpec.scala

```diff
@@ -0,0 +1,125 @@
+package io.iohk.scalanet.peergroup
+
+import java.nio.ByteBuffer
+import java.security.PrivateKey
+import java.security.cert.{Certificate, CertificateFactory}
+
+import io.iohk.decco.auto._
+import io.iohk.decco.BufferInstantiator.global.HeapByteBuffer
+import io.iohk.decco.{BufferInstantiator, Codec}
+import io.iohk.scalanet.NetUtils
+import io.iohk.scalanet.NetUtils.{aRandomAddress, isListeningUDP}
+import io.iohk.scalanet.TaskValues._
+import io.iohk.scalanet.peergroup.DTLSPeerGroup.Config._
+import monix.execution.Scheduler.Implicits.global
+import org.scalatest.FlatSpec
+import org.scalatest.Matchers._
+import org.scalatest.concurrent.ScalaFutures._
+import org.scalatest.RecoverMethods._
+
+import scala.concurrent.duration.Duration
+import scala.concurrent.duration._
+import scala.concurrent.{Await, Future}
+import scala.util.Random
+import DTLSPeerGroupSpec._
+import io.iohk.scalanet.peergroup.DTLSPeerGroup.Config
+
+class DTLSPeerGroupSpec extends FlatSpec {
+
+  implicit val patienceConfig = PatienceConfig(5 seconds)
+
+  behavior of "DTLSPeerGroup"
+
+  it should "send and receive a message" in {
```

Comment:
i it would probably be nice to have some negative test cases like:
what if there is not server listening on address that client tries to connect ?
what if one of peer disconect during peer dtls handshake ? 
what if one peer will have wrong keys and certifiicates ? 
what if peer won't support any of the ciphers ?

Reply:
Sure. A lot of this stuff will come in the future if/when we decide to use DTLS (e.g. for discovery).